merkletree/proof.go

```diff
@@ -88,13 +89,22 @@ func VerifyAuthPath(ap *AuthenticationPath,
 	return true
 }
 
+func VerifyCommitment(salt []byte, key string, value []byte, commitment []byte) bool {
+	got := crypto.Digest(salt, []byte(key), value)
```

Comment:
Btw, let's do this in a separate pull.  What you have here is fine for #57.  Just fix the typo.